DSA_Session5/src/employee/LinkedList.java

```diff
@@ -0,0 +1,69 @@
+package employee;
+/**
+ * program to get employee list based on their salary
+ * 
+ * @author Aman Gautam
+ * 
+ * Dated -06/08/2019 
+ */
+public class LinkedList {
+	Node headNode;
+
+	/**
+	 * method to insert data into linked list
+	 * 
+	 * @param list {LinkedList} contain employee's data
+	 * @param data of employee
+	 * @return list{List} of employee
+	 */
+	public LinkedList insert(LinkedList list, Employee data) {
+		Node newNode = new Node(data);
+		newNode.next = null;
+
+		if (list.headNode == null) {
+			list.headNode = newNode;
+		} else {
+			Node last = list.headNode;
+			while (last.next != null) {
+				last = last.next;
+			}
+			last.next = newNode;
+		}
+		return list;
+	}
+
+	/**
+	 * method to get list of employee sorted by salary
+	 * 
+	 * @param employeeList {LinkedList} list of employees (unsorted)
+	 * @return sorted list {LinkedList}
+	 */
+	public LinkedList sortBySalary(LinkedList employeeList) {
```

Comment:
Just call the sortBySalary(). No need to pass calling object in the sortBySalary() method.

Reply:
calling object removed . although changed the entire approach of method sortBySalary .